core/src/main/scala/org/apache/spark/storage/MemoryStore.scala

```diff
@@ -97,11 +117,11 @@ private[spark] class MemoryStore(blockManager: BlockManager, maxMemory: Long)
       returnValues: Boolean): PutResult = {
     if (level.deserialized) {
       val sizeEstimate = SizeEstimator.estimate(values.asInstanceOf[AnyRef])
-      val putAttempt = tryToPut(blockId, values, sizeEstimate, deserialized = true)
+      val putAttempt = tryToPut(blockId, () => values, sizeEstimate, deserialized = true)
       PutResult(sizeEstimate, Left(values.iterator), putAttempt.droppedBlocks)
     } else {
       val bytes = blockManager.dataSerialize(blockId, values.iterator)
-      val putAttempt = tryToPut(blockId, bytes, bytes.limit, deserialized = false)
+      val putAttempt = tryToPut(blockId, () => bytes, bytes.limit, deserialized = false)
```

Comment:
it's a little odd for this to be `() => bytes` everywhere. Can you create an alias that looks something like:
```
private def tryToPut(blockId: BlockId, value: Any, ...): ResultWithDroppedBlocks = {
  tryToPut(blockId, () => value(), ...)
}
```
same with dropFromMemory